R/pkg/inst/worker/daemon.R

```diff
@@ -30,8 +30,42 @@ port <- as.integer(Sys.getenv("SPARKR_WORKER_PORT"))
 inputCon <- socketConnection(
     port = port, open = "rb", blocking = TRUE, timeout = connectionTimeout)
 
+selectTimeout <- function() {
+  if (is.null(parallel:::selectChildren(timeout = 0))) {
+    # Wait a socket connection indefinitely if there are no workers running.
+    NULL
```

Comment:
Nice - this is in line with what I was thinking. One thing: Can we maintain some state in our own code to avoid calling `select` twice ? What I mean is if `finishedChildren` was `NULL` (based on [1]) then we set a flag saying `timeout <- NULL` -- next time we fork some children we set this back to `1`. Will that work or am I missing something ?

[1] https://stat.ethz.ch/R-manual/R-devel/library/parallel/html/children.html